pkg/controller/elasticsearch/nodespec/lifecycle_hook.go

```diff
@@ -23,31 +23,30 @@ const PreStopHookScript = `#!/usr/bin/env bash
 
 set -eux
 
-# This script will wait for up to $MAX_WAIT_SECONDS for $POD_IP to disappear from DNS record,
-# then it will wait additional $ADDITIONAL_WAIT_SECONDS and exit. This slows down the process shutdown
+# This script will wait for up to $PRE_STOP_MAX_WAIT_SECONDS for $POD_IP to disappear from DNS record,
+# then it will wait additional $PRE_STOP_ADDITIONAL_WAIT_SECONDS and exit. This slows down the process shutdown
 # and allows to make changes to the pool gracefully, without blackholing traffic when DNS
 # contains IP that is already inactive. Assumes $HEADLESS_SERVICE_NAME and $POD_IP env variables are defined.
 
 # max time to wait for pods IP to disappear from DNS. As this runs in parallel to grace period
-# (defaulting to 30s) after which process is SIGKILLed, it should be set to allow enough time
-# for the process to gracefully terminate.
-MAX_WAIT_SECONDS=${MAX_WAIT_SECONDS:=20}
+# after which process is SIGKILLed, it should be set to allow enough time for the process to gracefully terminate.
+PRE_STOP_MAX_WAIT_SECONDS=${PRE_STOP_MAX_WAIT_SECONDS:=20}
 
-# additional wait, allows queries to successfully use IP from DNS from before pod termination
-# this gives a little bit more time for clients that resolved DNS just before DNS record
-# was updated.
-ADDITIONAL_WAIT_SECONDS=${ADDITIONAL_WAIT_SECONDS:=1}
+# additional wait before shutting down Elasticsearch.
+# It allows kube-proxy to refresh its routes (defaults to every 30 seconds) and remove that Pod IP once Terminating.
+# Also gives some additional bonus time to in-flight requests to terminate.
```

Comment:
Not a strong opinion on this one, so I'm fine if you leave it like that, but I think it's a little bit imprecise, as this is not only about requests in-flight, but also new requests using IP from earlier DNS resolution.

Reply:
👍